### PR TITLE
add Titanic dataset

### LINKED ISSUE #90: add Titanic dataset
One of the most famous dataset for a beginner to start is the "Titanic dataset" , which is used for Exploratory data analysis and prediction of outcome using logistic regression, Decision trees, random forests, etc. I just think this dataset should be added so that can be easy for a beginner to get started in Machine learning in julia with a beginner friendly dataset. Also if approved, I am willing to work on this issue as it can be a great addition to these other famous datasets.

### PR DESCRIPTION
**Pull request by arcAman07**
In the src folder, created the Titanic Dataset folder where the csv file and the corresponding code is added.
In the tests folder , added the required tests for the three functions features,feature_names and targets. Code also updated in the runtime.jl file in the tests folder. Tested it in my local computer and it was working perfectly well. If this is merged, I will work on the corresponding documentation to go along with this dataset in the docs folder and writing better comments in my titanic.jl file in my titanic datasets folder. If any changes needed to be done just ping me and I will do it asap.

closes #90

### CODE REVIEW

File: test/tst_boston_housing.jl [hunk 1]
```diff
@@ -5,12 +5,12 @@ using MLDatasets
...
 @testset "Boston Housing" begin
-    X  = BostonHousing.features()
-    Y  = BostonHousing.targets()
+    X = BostonHousing.features()
+    Y = BostonHousing.targets()
     names = BostonHousing.feature_names()
     @test X isa Matrix{Float64}
     @test Y isa Matrix{Float64}
-    @test names == ["crim","zn","indus","chas","nox","rm","age","dis","rad","tax","ptratio","b","lstat"]
+    @test names == ["crim", "zn", "indus", "chas", "nox", "rm", "age", "dis", "rad", "tax", "ptratio", "b", "lstat"]
```
Comment (johnnychen94) [inline test/tst_boston_housing.jl]: These are unrelated changes.

**Review by johnnychen94 — APPROVED** [review]
I'd prefer to move `src/Titanic/titanic.csv` to `data/Titanic/titanic.csv` since it's not part of the source codes.

> If this is merged, I will work on the corresponding documentation to go along with this dataset in the docs folder and writing better comments in my titanic.jl file in my titanic datasets folder.

It can be as simple as adding a new file `docs/src/datasets/titanic.md` similar to
https://github.com/JuliaML/MLDatasets.jl/blob/master/docs/src/datasets/Iris.md and register it in `docs/make.jl`. Also with an entry to README.md.

**Comment by arcAman07** [comment]
Currently the titanic dataset is added. I f hopefully it is merged, I can start working on the documentation and get that done too asap

File: test/tst_boston_housing.jl (hunk 1 above)
Comment (arcAman07) [reply test/tst_boston_housing.jl]: Yeah actually i added some space to the boston housing test file feature names, so just was reverting back to the original.

Comment (arcAman07) [reply test/tst_boston_housing.jl]: Sure I will create a data folder and add the csv file there, I was just following the src code format of the boston housing market as it did quite similarly and I didn't want to write out of place code or change the sile strcuture of this library.

Comment (arcAman07) [reply test/tst_boston_housing.jl]: If a new data folder outside the src folder has to be created just confirm me I will do the same, I was just following the boston housing dataset and its code structure.

**Comment by johnnychen94** [comment]
> Sure I will create a data folder and add the csv file there, I was just following the src code format of the boston housing market as it did quite similarly and I didn't want to write out of place code or change the sile strcuture of this library.

Oh that's a good point, I didn't notice that boston housing case stores the data in `src/` folder

File: test/tst_boston_housing.jl [hunk 2]
```diff
@@ -5,8 +5,8 @@ using MLDatasets
 
 
 @testset "Boston Housing" begin
-    X  = BostonHousing.features()
-    Y  = BostonHousing.targets()
+    X = BostonHousing.features()
+    Y = BostonHousing.targets()
```
Comment (johnnychen94) [inline test/tst_boston_housing.jl:9]:
```suggestion
    X  = BostonHousing.features()
    Y  = BostonHousing.targets()
```

File: test/tst_boston_housing.jl (hunk 1 above)
Comment (arcAman07) [reply test/tst_boston_housing.jl]: Cool if this gets merged, if any help in documentation is needed I am surely willing to help.

File: src/Titanic/Titanic.jl [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+An interesting result may be obtained using functions from the Hmisc library.
+
+attach	(titanic3)
+plsmo	(age, survived, group=sex, datadensity=T) 
+# or group=pclass plot	(naclus	(titanic3)) # study patterns of missing values summary	(survived ~ age + sex + pclass + sibsp + parch, data=titanic3)
+
+"""
+
+module Titanic
```
Comment (johnnychen94) [inline src/Titanic/Titanic.jl:99]: we need

```suggestion
"""
module Titanic
```

so that it's linked to `?Titanic`.

**Comment by johnnychen94** [comment]
Let's add docs first and then we can make a new release.

File: test/tst_boston_housing.jl (hunk 1 above)
Comment (arcAman07) [reply test/tst_boston_housing.jl]: > Let's add docs first and then we can make a new release.

Sure so how do we plan that? Should I follow the custom template , make the docs and send the PR to get it reviewed or would you like to handle it?

File: src/Titanic/Titanic.jl [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+julia> using MLDatasets: Titanic
+
+julia> target = Titanic.targets();
+
+julia> summary(target)
+"1×891 Matrix{Float64}"
+     
+"""
+
+function targets(; dir = nothing)
```
Comment (johnnychen94) [inline src/Titanic/Titanic.jl:123]:
```suggestion
"""
function targets(; dir = nothing)
```

File: src/Titanic/Titanic.jl [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    reshape(Vector(titanic_data[2:end,2]), (1, 891))
+end
+
+"""
+    feature_names()
+
+Return the  the names of the features provided in the dataset.
+"""
+
+function feature_names()
```
Comment (johnnychen94) [inline src/Titanic/Titanic.jl:134]:
```suggestion
"""
function feature_names()
```

File: src/Titanic/Titanic.jl [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+julia> using MLDatasets: BostonHousing
+
+julia> features = BostonHousing.features();
+
+julia> summary(features)
+"13×506 Matrix{Float64}"
+```
+"""
+
+function features()
```
Comment (johnnychen94) [inline src/Titanic/Titanic.jl:155]:
```suggestion
"""
function features()
```

**Review by johnnychen94 — COMMENTED** [review]
Oh there're more docstring issues, don't forget to fix it in your following doc PR :)

**Comment by arcAman07** [comment]
> Oh there're more docstring issues, don't forget to fix it in your following doc PR :)

Yeah sure will just take a day's break as the entire day went for this this, but will do that asap. Thanks a lot for your help and see u soon
